calkit/models/pipeline.py

```diff
@@ -249,13 +252,76 @@ def dvc_outs(self) -> list[str | dict]:
         return outs
 
 
-class MatlabScriptStage(Stage):
+class MatlabStage(Stage):
+    def get_deps_from_matlab(self, filepath) -> list[str]:
+        """Automatically get dependencies from MATLAB.
+
+        If MATLAB cannot be invoked or returns an error, return the
+        declared stage inputs plus the provided filepath. When MATLAB
+        succeeds, return the discovered deps (converted to POSIX
+        relative paths) and ensure the provided filepath is included.
+        """
+        get_deps_cmd = self.xenv_cmd
+        if self.environment == "_system":
+            get_deps_cmd += "matlab -batch"
+        # Quote the filepath for MATLAB and escape single quotes
+        quoted = filepath.replace("'", "''")
+        # MATLAB code that adds the current folder to path and
+        # prints required files for the given file. Use disp to ensure
+        # newline-separated output.
+        matlab_code = (
+            "addpath(genpath(pwd)); "
+            f"f = matlab.codetools.requiredFilesAndProducts('{quoted}'); "
+            "for i=1:numel(f); disp(f{i}); end"
+        )
+
```

Comment:
There are just a few more blank lines in this method to delete, then we're good to go.